### ISSUE TITLE
Not possible to enable options for pyramids in raster "export > Save as..." when input is a WMS layer

### ISSUE THREAD

**Issue by gioman** [qgis/QGIS#53691]
### What is the bug or the crash?

![image](https://github.com/qgis/QGIS/assets/1951107/8e5a7db7-645b-4223-bfad-3aa7476701f8)

### Steps to reproduce the issue

Add a WMS layer and try to export it with "export > save as..."

The option for pyramids is not enabled by default and is not possible to enable it manually. As pyramids are added to the output, that is a local file on the file system, is not cleat why this options should not be available to users.

### Versions

3.28.8 3.32.0

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by agiudiceandrea** [comment]
Indeed, it seems the Pyramids group is enabled depending on the source layer driver capabilities instead of the destination layer driver capabilities:

https://github.com/qgis/QGIS/blob/9859ecbde74b534500fccacbd7163671f4004145/src/gui/qgsrasterlayersaveasdialog.cpp#L112-L133
